test/controllers/products_controller_test.rb

```diff
@@ -62,71 +118,87 @@
       must_redirect_to root_path
     end
 
-    it "must render to edit view if the merchant is logged in and there is a product" do
+    it "must show edit view if the merchant is logged in and there is a product" do
       @merchant = merchants(:sappy1)
       login(@merchant, :github)
 
       get edit_product_path(one.id)
       must_respond_with :success
     end
-
   end
 
-  describe "update" do
-    let(:one) {products(:tree1)}
-    let(:merchant1) {merchants(:sappy1)}
-    let(:merchant2) {merchants(:sappy2)}
+  describe "update a product with a valid merchant signed in" do
+      let(:merchant1) {merchants(:sappy1)}
+      let(:one) {products(:tree1)}
 
-    it "will redirect if the product is invalid" do
-      patch product_path(-1), params: { categories: ["Birthday"] }
-      flash[:status].must_equal :error
-      flash[:result_text].must_equal "That is not a valid product"
-      must_redirect_to root_path
+    before do
+      login(merchant1, :github)
     end
 
-    it "will not save and will redirect if the merchant of the product is not signed in" do
-      #no merchant signed in
-      categories = one.categories
+    it "successfully updates the categories of the product when selected" do
+      session[:merchant_id].to_i.must_equal merchant1.id
 
-      patch product_path(one.id), params:{categories: ["Birthday"]}
-      flash[:status].must_equal :error
-      must_redirect_to root_path
+      #format passed through params if chosen from a list
+      before = one.categories.count
```

Comment:
could be a good spot for a proc{ someblock }.must_change

Reply:
Maybe we can look at that one together.. I wanted to be able to do so while also checking that the categories includes the newly added category.